components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/AwesomeBarFeature.kt

```diff
@@ -172,17 +172,21 @@ class AwesomeBarFeature(
      * @param loadUrlUseCase the use case invoked to load the url when the user clicks on the suggestion.
      * @param engine optional [Engine] instance to call [Engine.speculativeConnect] for the
      * highest scored suggestion URL.
+     * @param maxNumberOfSuggestions optional parameter to specify the maximum number of returned suggestions.
+     * Zero or a negative value here means the default number of history suggestions will be returned.
      */
     fun addHistoryProvider(
         historyStorage: HistoryStorage,
         loadUrlUseCase: SessionUseCases.LoadUrlUseCase,
         engine: Engine? = null,
-        maxNumberOfResults: Int = -1
+        maxNumberOfSuggestions: Int = -1
     ): AwesomeBarFeature {
         awesomeBar.addProviders(
-            HistoryStorageSuggestionProvider(
-                historyStorage, loadUrlUseCase, icons, engine, maxNumberOfResults
-            )
+            if (maxNumberOfSuggestions <= 0) {
```

Comment:
Instead of setting `maxNumberOfSuggestions` to zero to not receive history suggestions clients should not `addHistoryProvider`.